include/adm/detail/named_type_validators.hpp

```diff
@@ -23,6 +24,20 @@ namespace adm {
   };
 
   namespace detail {
+    template <typename T>
+    void validateStringList(std::string const& type, std::string const& value,
+                            T&& list) {
```

Comment:
Perhaps change this to `std::initializer_list<std::string> list`? That way this doesn't get instantiated for all the different array lengths, and callers don't have to write the type (it would look like `validateStringList("HorizontalEdge", value, {"left", "right"});`)

It might be better to accept `const char *`s instead to avoid allocations, as this ultimately gets used in the xml position parser.

Reply:
f92055e8c1e10a826fa49ddb473b7cb85cb63db5 addresses this